### ISSUE TITLE
coc-nvim: Unknown function `coc:float#createNotification`

### ISSUE THREAD

**Issue by RasmusRendal** [NixOS/nixpkgs#176753]
### Steps To Reproduce
1. In your `configuration.nix`, using 22.05, enable `neovim` as follows:
```nix
{
    programs.neovim = {
        enable = true;
        withNodeJs = true;
        configure.packages.myVimPackages.start = with pkgs.vimPlugins; [ coc-nvim ];
    };
}
```

### Build log
```
building '/nix/store/ajzf0zdmsyb49r8d2l1zb934jwf9nys0-neovim-0.7.0.drv'...
Generating remote plugin manifest
Error detected while processing function coc#rpc#start_server[13]..coc#util#get_data_home:
line   15:
E117: Unknown function: coc#float#create_notification
remote/host: generated rplugin manifest: /nix/store/rxgr41adixzjvq3730xv7814i2s8a7n9-neovim-0.7.0/rplugin.vim
Generating rplugin.vim failed!
error: builder for '/nix/store/ajzf0zdmsyb49r8d2l1zb934jwf9nys0-neovim-0.7.0.drv' failed with exit code 1
```

### Additional context
The problem was not present in 21.11.

### Notify maintainers
@manveru @rvolosatovs @ma27

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.43, NixOS, 22.05 (Quokka), 22.05.583.ca2629644d4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.05, unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by RasmusRendal** [comment]
As a workaround, you can put `coc-nvim` in opt, and then load it in your vimrc:
```nix
{
    programs.neovim = {
        enable = true;
        withNodeJs = true;
        configure.packages.myVimPackages.opt = with pkgs.vimPlugins; [ coc-nvim ];
        customRC = ":packadd coc.nvim";
    };
}
```

**Comment by RossComputerGuy** [comment]
@RasmusRendal solution does not work with home-manager.

**Comment by abathur** [comment]
Closing since it sounds like the issue was fixed in master/unstable and 22.05 is now EOL (making the backport moot...).

Happy to reopen if you think I'm overreaching.
